ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsIngestion.scala

```diff
@@ -88,4 +93,46 @@ private[backend] trait StorageBackendTestsIngestion
     partiesAfterLedgerEndUpdate should not be empty
   }
 
+  private val NumberOfUpsertPackagesTests = 100
+  it should s"safely upsert packages concurrently ($NumberOfUpsertPackagesTests)" in withConnections(
+    2
+  ) { connections =>
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    def test() = {
+      val List(connection1, connection2) = connections
```

Comment:
Would you mind moving common pieces out of the `test()` method to make it easier to read? E.g.:
```
    def test() = {
      executeSql(backend.parameter.initializeParameters(someIdentityParams))
      
      val ingestF1 = ingestPackagesF(connection1, packages1 ++ conflictingPackageDtos)
      val ingestF2 = ingestPackagesF(connection2, packages2 ++ conflictingPackageDtos)
      
      Await.result(ingestF1, 10.seconds)
      Await.result(ingestF2, 10.seconds)

      executeSql(updateLedgerEnd(offset(50), 0))

      executeSql(backend.packageBackend.lfPackages).keySet.map(_.toString) shouldBe (
        conflictingPackageDtos ++ packages1 ++ packages2
      ).map(_.package_id).toSet
    }
 ```